### ISSUE TITLE
East Palo Alto Front End Questions / Comments

### ISSUE THREAD

**Issue by MarkusElKatsha** [CityScope/CS_cityscopeJS#171]
Following is list of question and comments for the (information panel on the right hand side (radar plot, doughnut plot etc..):

1) Area values in the shown on the doughnut  chart should be in square meters or square feet.
2) Scale on bar charts should be in percentile (not 0 - 1)
3) Radar and Bar Charts are not updating. We need two weeks to play / debug the tool.

**Comment by MarkusElKatsha** [comment]
Following is a list of questions/comments for the main display (over the satellite image):
1. Heat map transparency - add sider to adjust the transparency of heat maps / overlays
2. The reddest zones of the heat map are the most transparent, yellow to green areas become opaque (see attached image). 
3. Names of heat maps require classification (is the "Housing" visualization referring to density, accessibility or diversity.
4. Can a key be added to each heat map scenario? (what does red mean? What does green mean?)
5. Housing Heat Map (is all green). What is the message?
6. Retail Heat map is partially green. What is the message?
7. Jobs Heat map is all green. What is the message?


![Screen Shot 2023-01-17 at 3 21 19 PM](https://user-images.githubusercontent.com/35349500/213004922-b043eda2-618f-4c10-93dd-69a2a62f5d98.png)

**Comment by MarkusElKatsha** [comment]
Trips Volume Visualizations are unclear: 
1. path colors should show congestion, not mobility choice
2. In the congestion map all roads should show a color from (green to red)
3. If the thickness of the road indicates volume, add a key or  legend explaining thickness. 
4. Can and ABM visualization be used? (similar to attached video clip)


 

https://user-images.githubusercontent.com/35349500/213010715-a385508b-5e56-4114-a220-4086aac6315b.mp4

**Comment by RELNO** [comment]
@MarkusElKatsha thanks for your comments. In the future, please split issues to a single item/issue so I can handle/close each issue separately. thanks. 

> Area values in the shown on the doughnut chart should be in square meters or square feet.

most CS project have areas that are in the +100k sq/m per type, this will make it extremely large numbers. We can do Exponential Notation https://www.ibm.com/docs/en/zvm/7.2?topic=arithmetic-exponential-notation, but I don't think many people are familiar with them  

> Scale on bar charts should be in percentile (not 0 - 1)

Doable, though confusing since each bar is not part of a 100% with the rest of bars, which might be implied with percentage.  

> Radar and Bar Charts are not updating. We need two weeks to play / debug the tool.  

Not a CSjs issue. Probably backend modules are off ATM.

**Comment by RELNO** [comment]
> Heat map transparency - add sider to adjust the transparency of heat maps / overlays
> The reddest zones of the heat map are the most transparent, yellow to green areas become opaque (see attached image).

The core issue here is that this is not a real heatmap, but a grid with values (=circle sizes) not based on density of points. 
(for ref., that's a `real` heatmap: https://deck.gl/docs/api-reference/aggregation-layers/heatmap-layer zoom in to see the density)
I'm more than happy to go with a simple grid/hex based heatmap (like we do on the table in Volpe/Ukraine) @doorleyr what do you think? 

> Names of heat maps require classification (is the "Housing" visualization referring to density, accessibility or diversity.

Names are driven from the module backend, @doorleyr can edit them to your liking. We can explore a subtitle with more info if needed, but again - this comes from the backend. 

> Can a key be added to each heat map scenario? (what does red mean? What does green mean?)

See my prev. answer, we can add a text snippet to explain the range in each heatmap 

> Housing Heat Map (is all green). What is the message?
> Retail Heat map is partially green. What is the message?
> Jobs Heat map is all green. What is the message?

[These are three great questions. ](https://gfycat.com/closedpeskygander-mysteries-maverick-top-gun-mystery-cruise-lifes)

**Comment by RELNO** [comment]
> Trips Volume Visualizations are unclear:

> path colors should show congestion, not mobility choice

I'm not sure I understand: path colors show the volume (=# of trips) of certain mode/profile in their most popular routes

> In the congestion map all roads should show a color from (green to red)

That is a different visualization than the one we have. trip volume is a not how congested a road is, rather it's what mode runs where and how much. In practice: 1 -  I don't know if @doorleyr is computing congestion on the fly. 2 - we'll need to add a new layer for that and test it. 

> If the thickness of the road indicates volume, add a key or legend explaining thickness.

Sure, care to draft a subtitle/helper? This can be done the same as in the area calculations 
<img width="294" alt="Screenshot 2023-01-17 at 5 23 00 PM" src="https://user-images.githubusercontent.com/14296774/213025461-052be177-16b1-4488-b688-044cf46b0221.png">


> Can and ABM visualization be used? (similar to attached video clip)

As you might recall, we had these 'moving agents' thingy for some time, but was removed due to irrelevancy and performance. 

The common feedback was that, besides being visually nice, it adds very little info (agent #345345 moved from here to there via this route, what did we learned from it?) 

From a practical standpoint, animated agents is a huge overload on the system (+1M draw calls/frame in complex scene), and weaker machines (which we must consider will be in use by Emerson/EPA ppl) tend to become laggy when it's on. The video you show animates very few agents for a pretty massive site, but seems to already run at a frame rate 2fps.
